ion_schema_2_0/open_content/user_fields_in_schema_footer.isl

```diff
@@ -0,0 +1,130 @@
+$ion_schema_2_0
+
+$test::{
+  description: "Unreserved symbols should always be permitted as user field names in a schema footer",
+  valid_schemas: [
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ '': 0 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ $lower_snake_case_with_a_leading_dollar_sign: 1 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ _lower_snake_case_with_leading_underscore: 2 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ SCREAMING_SNAKE_CASE: 3 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ 'UPPER-KEBAB-CASE': 4 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ 'lower-kebab-case': 5 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ camelCase: 6 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ 'name with whitespace': 7 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ '🥧_foo': 8 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ 'é∑å®ê': 9 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ _: 10 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ $: 11 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ 'éomer': 12 }
+    ),
+    (
+      $ion_schema_2_0
+      schema_header::{}
+      schema_footer::{ $ion_schemabutnotreally: 13 }
+    ),
+  ]
+}
+
+$test::{
+  description: "Reserved symbols that are not declared as user fields should not be permitted as user field names in the schema footer",
+  isl_for_isl_can_validate: false,
```

Comment:
Ion Schema Schemas (also known as isl_for_isl) cannot catch this type of invalid schema because ISL cannot (and is not intended to) perform validation that is dependent on other parts of the data. Basically, it is not possible write an Ion Schema that can detect if an arbitrary field in a type definition had been declared in the `user_fields` section.

Reply:
> Ion Schema Schemas (also known as isl_for_isl) cannot catch this type of invalid schema because ISL cannot (and is not intended to) perform validation that is dependent on other parts of the data. Basically, it is not possible write an Ion Schema that can detect if an arbitrary field in a type definition had been declared in the `user_fields` section.

Thanks for the explanation.